walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -100,26 +102,27 @@
   */
   private ArrayList<TimeRange> mergeTimeRangesThatIntersect(ArrayList<TimeRange> optional, ArrayList<TimeRange> mandatory) {
     ArrayList<TimeRange> finalTimeRanges = new ArrayList<>();
-
-    ArrayList<Integer> startValues = new ArrayList<>();
-    for (TimeRange mandatoryTimeRange: mandatory) {
-      startValues.add(mandatoryTimeRange.start());
-    }
+    int mandatoryIndex = 0;
 
     for (TimeRange optionalTimeRange: optional) {
       int start = optionalTimeRange.start();
       int end = optionalTimeRange.end();
 
-      // This only works because java's binary search returns (-(insertion_point) - 1) 
-      // if not found, meaning that the negative is 2 indexes beyond what the closest 
-      // lesser than element exists
-      int index = Collections.binarySearch(startValues, start);
-      index = index >= 0 ? index : (index + 2) * -1;
-  
-      if (index != -1) {
-        int mandatoryEnd = mandatory.get(index).end();
-        if (end <= mandatoryEnd) {
+      while (mandatoryIndex < mandatory.size()) {
+        TimeRange possibleIntersection = mandatory.get(mandatoryIndex);
+        
+        int mandatoryStart = possibleIntersection.start();
+        int mandatoryEnd = possibleIntersection.end();
+
+        if (mandatoryStart > start) {
```

Comment:
It's not totally clear what's expected, but based on [this section of the walkthrough](https://github.com/Domene99/STEP/blob/master/walkthroughs/week-5-tdd/project/calendar-walkthrough.md#writing-more-tests): 

> Also based on everyAttendeeIsConsidered, add an optional attendee C who has an event between 8:30 and 9:00. Now only the early and late parts of the day should be returned.

This seems to suggest that if there exist time ranges that work for all attendees, then the method should only return time ranges that work for all attendees. In the example you provided, that means the correct return value is `|3 4|`

Reply:
Running this test manually i do get | 3 4 |, as when the time range gets added in L122 it immediately breaks and goes out of bounds for the optional arraylist. Besides all tests do seem to be working just like they were with the previous algorithm.
You are indeed correct that it only uses time ranges from optional this is due to the fact that what I'm checking is that for every optional time range is there a available mandatory time range during this same period of time. This means that for this optional time range all mandatory and optional  attendees can attend. I'm not sure I'm explaining it as best as possible
Ex:
// Events   :   |-B(Opt)-|
// Day       : |-----------------------|
// Options : |-|             |----------|

Reply:
okay I can now see that the example did not keep the spacing I gave it. Basically imagine an event that splits the day into two

Reply:
Hmm, I'm pretty sure I'm missing something here. 

This is what I thought would happen:
L108: `start = 1`
L113: `mandatoryStart = 3`
L117: `mandatoryStart > start`
L118: `break`
L107: no more optional elements, loop ends
L130: return `mandatory`


But I trust your test case more than I trust me just reading code, so I'm gonna approve. Let me know if you see where my misunderstanding is though.

Reply:
Ohhhhh. The misunderstanding was from my part. I didn't see it as one time range. I saw it as |1-2| and then |3-4|. You're right when you say I should be merging them instead of just choosing one. I'll add this test case and fix it